internal/queue/consumer.go

```diff
@@ -57,21 +61,36 @@ func (q *RabbitQueue) StartConsumer(queueData ConsumerQueue, worker worker.Worke
 	if !ok {
 		return errors.New("consumer not found")
 	}
-	go func() {
-		for message := range messages {
-			err := worker.Process(message.Body)
+
+	errGroup := new(errgroup.Group)
+	for message := range messages {
+		errGroup.Go(func() error {
+			cbWorker, ok := queueData.RoutingKeys[message.RoutingKey]
+			if !ok {
+				return errors.New(fmt.Sprintf("%s not implemented as valid routing key", message.RoutingKey))
+			}
+
+			err := cbWorker.Process(message.Body)
+
 			if err != nil {
 				err = message.Acknowledger.Nack(message.DeliveryTag, false, true)
 				if err != nil {
-					panic(err) // I know this isn't good. Will need to fix it
+					return errors.New(err.Error())
```

Comment:
We should probably add more context to these then just wrapping error in a new error